### ISSUE TITLE
Plan to release etcd v3.5.18

### ISSUE THREAD

**Issue by ivanvc** [etcd-io/etcd#19119]
### What would you like to be added?

The etcd [patch release criteria](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/release.md#patch-release-criteria) has been met for our `release-3.5` stable release branch so we should release `v3.5.18`.

#### The list of commits included since the previous release is: [v3.5.17...release-3.5](https://github.com/etcd-io/etcd/compare/v3.5.17...release-3.5):

- #19113
- #18899 
- #18829 
- #18937 
- #18999 
- #19004
- #19145 
- #19113 
- #19154
- #19161
- #19158 
- #19164 
- #19167 
- #19211  
- #19249
- #19258  

Work in progress CHANGELOG is: https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v3518-tbc

#### List of pull requests we still need to backport from `main` to `release-3.5` before the patch release is issued:

- [x] https://github.com/etcd-io/etcd/pull/19043
  - #19154
- [x] https://github.com/etcd-io/etcd/pull/19101
  - #19158

#### Release team

| GitHub handle | Role |
|---|---|
| @jmhbnz | Release Lead |
| @ivanvc | Release Advisor |
| @ghouscht | Release Shadow |

### Why is this needed?

Regular patch releases are vital to ensure our users have bug-free and secure software.

**Comment by ivanvc** [comment]
I'd like to be a release advisor/shadow, as I'd like to see someone else running the release script after the first improvements from #18604. However, I can also be the lead if there are no volunteers.

**Comment by ghouscht** [comment]
I haven't had much time in the last few weeks to help with etcd development but I think I could once again be a shadow for the release. I think I'm not yet familiar enough with the process to be a lead.

**Comment by taraspos** [comment]
@ivanvc would it be possible to include patches to `x/crypto` and `x/net` to resolve following vulnerabilities?
- https://github.com/advisories/GHSA-v778-237x-gjrc
- https://github.com/advisories/GHSA-w32m-9786-jp63

**Comment by cwayne18** [comment]
@taraspos looks like `x/net` was bumped here: https://github.com/etcd-io/etcd/commit/bf3bb6d8beed85922e900c65fe85c7dfb4840055

**Comment by taraspos** [comment]
Yeah, but it needs to be backported into the [release-3.5](https://github.com/etcd-io/etcd/tree/release-3.5) branch to get into the 3.5.x release as far as I understand:
- https://github.com/etcd-io/etcd/blob/release-3.5/go.mod#L87-L88

**Comment by cwayne18** [comment]
oof, of course you're correct, sorry, still seems i have holiday-brain. It would certainly be good to have those fixes included for us

**Comment by ivanvc** [comment]
> @ivanvc would it be possible to include patches to `x/crypto` and `x/net` to resolve following vulnerabilities?
> 
>     * [GHSA-v778-237x-gjrc](https://github.com/advisories/GHSA-v778-237x-gjrc)
> 
>     * [GHSA-w32m-9786-jp63](https://github.com/advisories/GHSA-w32m-9786-jp63)

This makes sense. We're not directly affected by the vulnerability (https://github.com/etcd-io/etcd/actions/runs/12551556226/job/34996167337), but it's still a dependency with a CVE, and according to our [Dependency Management documentation](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#stable-branches).

**Comment by ivanvc** [comment]
@jmhbnz, would you be interested in being part of the release team?

**Comment by jmhbnz** [comment]
> @jmhbnz, would you be interested in being part of the release team?

Thanks for the reminder on this, yes we need to get this release cut, we could aim to do it later this week perhaps.  

I'll take lead for this one.  Are you available to be a second pair of eyes for me as release advisor @ivanvc?

**Comment by ivanvc** [comment]
> I'll take lead for this one. Are you available to be a second pair of eyes for me as release advisor @ivanvc?

Absolutely. Looks like we have a team. I'll groom to see if there are other potential PRs to backport. We also need to merge #19113 first.

**Comment by ivanvc** [comment]
It would be good to merge #19122 (blocked by DCO/CI) or manually edit the release notes when publishing the release.

**Comment by ahrtr** [comment]
> Absolutely. Looks like we have a team. I'll groom to see if there are other potential PRs to backport. We also need to merge #19113 first.

Merged.



> @ivanvc would it be possible to include patches to `x/crypto` and `x/net` to resolve following vulnerabilities?
> 
> * [GHSA-v778-237x-gjrc](https://github.com/advisories/GHSA-v778-237x-gjrc)
> * [GHSA-w32m-9786-jp63](https://github.com/advisories/GHSA-w32m-9786-jp63)

It should a quick fix. can anyone deliver a PR?



> It would be good to merge #19122 (blocked by DCO/CI) or manually edit the release notes when publishing the release.

Since there is no response from the contributor. Please feel free to fix it yourself.

**Comment by ivanvc** [comment]
> It should a quick fix. can anyone deliver a PR?

I'm on it.

**Comment by ivanvc** [comment]
> Since there is no response from the contributor. Please feel free to fix it yourself.

@ahrtr, ah, I didn't see this comment. I first replied on https://github.com/etcd-io/etcd/pull/19122#issuecomment-2581096611. I'll do it, coauthoring with the original contributor.

**Comment by serathius** [comment]
Please postpone release until we root cause https://github.com/etcd-io/etcd/issues/19179

**Comment by ivanvc** [comment]
Also, Go 1.22.11 is being released this Thursday 16th. So, it would be good to release the latest version.

**Comment by ivanvc** [comment]
Raised #19211 to bump Go to 1.22.11

**Comment by ivanvc** [comment]
@jmhbnz, with #19179 closed and no outstanding tasks, we should be ready to schedule the release. Can you send a new calendar invite? Thanks

**Comment by jmhbnz** [comment]
etcd `v3.5.18` has been released today: https://github.com/etcd-io/etcd/releases/tag/v3.5.18

Action items from release meeting:
- @ivanvc to raise pr updating release script to automatically publish github release rather than publish as draft. This functionality has been well tested now and is working great.
- @jmhbnz to raise pr proposing updating release workflow to use @etcd-io/release-etcd team to bypass branch restrictions rather than manually turning off and on branch restrictions during release window.  During normal activity the @etcd-io/release-etcd team will have no members, during a release the group will be populated with the release team members who will then automatically be able to bypass branch restricutions to carry out the release.  After the release the team will be reverted to no members.

**Comment by fuweid** [comment]
Thanks @jmhbnz @ivanvc !

**Comment by ivanvc** [comment]
Hi @jmhbnz, can we revert now https://github.com/kubernetes/org/pull/5360? Or is there a particular reason to maintain the elevated permissions?

**Comment by jmhbnz** [comment]
> Hi [@jmhbnz](https://github.com/jmhbnz), can we revert now [kubernetes/org#5360](https://github.com/kubernetes/org/pull/5360)? Or is there a particular reason to maintain the elevated permissions?

We can revert now, opened https://github.com/kubernetes/org/pull/5376.  Wanted to ensure #19270 was closed off before reverting. Thanks!

**Comment by taraspos** [comment]
Hey @ivanvc @jmhbnz seems like [CHANGELOG](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v3518-2025-01-24) is missing mentions for following dependency upgrades:
- https://github.com/etcd-io/etcd/pull/19154
- https://github.com/etcd-io/etcd/pull/19158

**Comment by ivanvc** [comment]
> Hey @ivanvc @jmhbnz seems like [CHANGELOG](https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#v3518-2025-01-24) is missing mentions for following dependency upgrades:
> - https://github.com/etcd-io/etcd/pull/19154
> - https://github.com/etcd-io/etcd/pull/19158

Good point, do you want to open a PR? Else, I can do it tomorrow.

**Comment by ivanvc** [comment]
> @ivanvc to raise pr updating release script to automatically publish github release rather than publish as draft. This functionality has been well tested now and is working great.

Due to the issues with the latest 3.6.0-rc.0 release. I'm going to park this. I'm also reconsidering how to set the latest release and pre-release flags for our now three active branches.

I'll close this issue as without that outstanding action item there's nothing else to do.
